internal/providers/pluginfw/tfschema/struct_to_schema.go

```diff
@@ -32,6 +33,7 @@ func typeToSchema(v reflect.Value) NestedBlockObject {
 		}
 		isOptional := fieldIsOptional(typeField)
 		isComputed := fieldIsComputed(typeField)
+		isObject := fieldIsAnObject(typeField)
```

Comment:
Implementation suggestion: we should define a struct that captures all elements of the struct tag:
```go
type structTag struct {
        optional bool
        computed bool
        singleObject bool
}
```
Then we have one method to extract the struct tag. This means less plumbing of the struct attributes.

The other huge advantage of this is that it is a single, natural place where we can document all supported struct tags.

Reply:
Great idea, thanks!